examples/step-69/step-69.cc

```diff
@@ -0,0 +1,2893 @@
+    DEAL_II_ALWAYS_INLINE inline Tensor<1, k> gather(const std::array<T1, k> &U,
+                                                     const T2                 i)
+    {
+      Tensor<1, k> result;
+      for (unsigned int j = 0; j < k; ++j)
+        result[j] = U[j].local_element(i);
+      return result;
+    }
+
+    // <code>scatter()</code>: this function has three input arguments, the
+    // first one is meant to be a "global object" (say a locally owned or
+    // locally relevant vector), the second argument which could be a
+    // <code>Tensor<1,problem_dimension></code>, and the last argument
+    // which represents a index of the global object. This function will be
+    // primarily used to write the updated nodal values, stored as
+    // <code>Tensor<1,problem_dimension></code>, into the global object.
+
+    template <typename T1, std::size_t k1, typename T2, typename T3>
+    DEAL_II_ALWAYS_INLINE inline void
+    scatter(std::array<T1, k1> &U, const T2 &result, const T3 i)
+    {
+      for (unsigned int j = 0; j < k1; ++j)
+        U[j].local_element(i) = result[j];
+    }
+  } // namespace
+
+  // We are now in a position to assemble all matrices stored in
+  // <code>OfflineData</code>: the lumped mass entries $m_i$, the
+  // vector-valued matrices $\mathbf{c}_{ij}$ and $\mathbf{n}_{ij} =
+  // \frac{\mathbf{c}_{ij}}{|\mathbf{c}_{ij}|}$, and the boundary normals
+  // $\boldsymbol{\nu}_i$.
+  //
+  // In order to exploit thread parallelization we use WorkStream approach
+  // detailed in the @ref threads Parallel computing with multiple processors
+  // accessing shared memory. As customary this requires
+  // definition of
+  //  - Scratch data (i.e. input info required to carry out computations): in
+  //    this case it is <code>scratch_data</code>.
+  //  - The worker: in the case it is <code>local_assemble_system()</code> that
+  //    actually computes the local (i.e. current cell) contributions from the
+  //    scratch data.
+  //  - A copy data: a struct that contains all the local assembly
+  //    contributions, in this case <code>CopyData<dim>()</code>.
+  //  - A copy data routine: in this case it is
+  //    <code>copy_local_to_global()</code> in charge of actually coping these
+  //    local contributions into the global objects (matrices and/or vectors)
+  //
+  // Most the following lines are spent in the definition of the worker
+  // <code>local_assemble_system()</code> and the copy data routine
+  // <code>copy_local_to_global()</code>. There is not much to say about the
+  // WorkStream framework since the vast majority of ideas are reasonably
+  // well-documented in step-9, step-13 and step-32 among others.
+  //
+  // Finally, assuming that $\mathbf{x}_i$ is a support point at the boundary,
+  // the (nodal) normals are defined as:
+  //
+  // @f{align*}
+  // \widehat{\boldsymbol{\nu}}_i :=
+  // \frac{\boldsymbol{\nu}_i}{|\boldsymbol{\nu}_i|} \ \text{where}
```

Comment:
You need a bit of space:
```suggestion
  // \frac{\boldsymbol{\nu}_i}{|\boldsymbol{\nu}_i|} \ \text{ where }
```